lib/salus/scanners/gitleaks.rb

```diff
@@ -0,0 +1,109 @@
+require "salus/scanners/base"
+require "json"
+
+# Gitleaks scanner integration. Detects hardcoded secrets like passwords,
+# api keys, and tokens in git repos.
+# https://github.com/zricethezav/gitleaks
+
+module Salus::Scanners
+  class Gitleaks < Base
+    def run
+      Dir.chdir(@repository.path_to_repo) do
+        copts = config_options
+        shell_return = run_shell(
+          'gitleaks '\
+          '--path . '\
+          '--format sarif '\
+          '--report /dev/stdout '\
+          '--leaks-exit-code 0 ' +
+          copts
+        )
+
+        report_stderr(shell_return.stderr)
```

Comment:
`report_stderr` should only be called if there is actually an error.

Reply:
Gitleaks reports runtime stats to stderr. Ex:
```
INFO[0001] scan time: 1 second 870 milliseconds 717 microseconds
WARN[0001] leaks found: 14
```
Is there a better place to report this information?

Reply:
What if 0 leaks found? Will Gitleaks still report something like below to stderr?
```
INFO[0001] scan time: 1 second 870 milliseconds 717 microseconds
WARN[0001] leaks found: 0
```

Reply:
We don't really need to report the runtime info.